processor/metric/generated/schema/payload.go

```diff
@@ -0,0 +1,228 @@
+package schema
+
+const PayloadSchema = `{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "$id": "docs/spec/metrics/payload.json",
+    "title": "Metrics payload",
+    "description": "Metrics for correlation with other APM data",
+    "type": "object",
+    "properties": {
+        "metrics": {
+            "type": "array",
+            "items": {
+                    "$schema": "http://json-schema.org/draft-04/schema#",
+    "$id": "docs/spec/metrics/metric.json",
+    "type": "object",
+    "description": "Metric data captured by an APM agent",
+    "properties": {
+        "samples": {
+            "type": ["object", "null"],
+            "description": "Sampled application metrics collected from the agent",
+            "regexProperties": true,
+            "patternProperties": {
+                "^[^*\"]*$": {
+                        "$schema": "http://json-schema.org/draft-04/schema#",
+    "$id": "docs/spec/metrics/sample.json",
+    "type": ["object", "null"],
+    "description": "A single metric sample.",
+    "oneOf": [
+        {
+            "properties": {
+                "type": {"enum": ["counter", "gauge"]},
+                "unit": {"type": ["string", "null"]},
```

Comment:
A `description` about what the unit affects would be helpful. For example does it influence the way Kibana formats the values? Which are the special values here? For example `byte` vs. `bytes`, `kb` vs. `KiB`.

Reply:
Good idea, lacking in descriptions in general here.